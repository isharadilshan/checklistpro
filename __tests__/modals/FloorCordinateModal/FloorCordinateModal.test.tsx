import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import FloorCordinateModal from '../../../src/components/modals/FloorCordinateModal';

configure({adapter: new Adapter()});

const props = {
  isModalVisible: true,
  closeModal: jest.fn(),
  cordinates: {
    xCordinate: 120,
    yCordinate: 30,
  },
};

describe('when rendering', () => {
  const wrapper = shallow(<FloorCordinateModal {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
