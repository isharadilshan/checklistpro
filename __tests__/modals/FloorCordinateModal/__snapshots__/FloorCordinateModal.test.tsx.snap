// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<ModalWrapper
  closeModal={[MockFunction]}
  modalVisible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "padding": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "padding": 30,
        }
      }
    >
      <Title
        style={
          {
            "color": "#BBC8D6",
          }
        }
      >
        X CORDINATE:   120
      </Title>
      <Title
        style={
          {
            "color": "#BBC8D6",
          }
        }
      >
        Y CORDINATE:   30
      </Title>
    </View>
    <Button
      label="OK"
      mode="contained"
      onPress={[MockFunction]}
      style={
        {
          "width": 200,
        }
      }
    />
  </View>
</ModalWrapper>
`;
